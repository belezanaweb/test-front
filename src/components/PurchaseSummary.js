/** @jsx jsx */
import { jsx } from '@emotion/core'
import styled from '@emotion/styled'
import { PropTypes } from 'prop-types';

const Summary = styled('section')`
  background-color: transparent;
	padding: 14px;
	margin-top: 20px;
	border-radius: 3px;
	min-width: 285px;
  border: solid 1px #D6D6D6;
  
  div {
    display: flex;
    justify-content: space-between;
    margin-top: 9px;
    font-size: 14px;
  }

  .first-child {
    margin-top: 0;
  }
  .last-child {
    margin-top: 16px;
  }
  .orange {
    color: #FF7800;
  }

  .total{
    font-size: 14px;
    font-weight: 700;
    line-height: 17px;
  }
`
const PurchaseSummary = ({shoppingCart}) => {
    const format = { minimumFractionDigits: 2 , style: 'currency', currency: 'BRL' }
    const subTotal = shoppingCart.discount && (shoppingCart.subTotal).toLocaleString('pt-BR', format)
    const shippingTotal = shoppingCart.discount && (shoppingCart.shippingTotal).toLocaleString('pt-BR', format)
    const discount = shoppingCart.discount && (shoppingCart.discount).toLocaleString('pt-BR', format)
    const total = shoppingCart.discount && (shoppingCart.total).toLocaleString('pt-BR', format)

    return(
        <Summary>
            <div className='first-child'>
                <span>PRODUTOS</span>
                <span>{subTotal}</span>
            </div>
            <div>
                <span>FRETE</span>
                <span>{shippingTotal}</span>
            </div>
            <div className='orange'>
                <span>DESCONTO</span>
                <span>- {discount}</span>
            </div>
            <div className='last-child'>
                <span className='total'>TOTAL</span>
                <span className='total'>{total}</span>
            </div>
        </Summary>
    )
}


PurchaseSummary.propTypes = {
    shoppingCart: PropTypes.object.isRequired,
}

export default PurchaseSummary