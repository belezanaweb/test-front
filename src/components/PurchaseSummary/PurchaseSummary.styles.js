import styled from 'styled-components'

export const PurchaseSummaryWrapper = styled.section`
  display: grid;
  grid-gap: 9px;
  padding: 15px 13px 13px 14px;
  border: 1px solid #ccc;
  border-radius: 3px;
  color: #212122;
  font-size: 14px;
  line-height: 1.21428571428571;
`

export const PurchaseSummaryList = styled.dl`
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  margin: 0;
`

export const PurchaseSummaryTerm = styled.dt`
  text-transform: uppercase;
`

export const PurchaseSummaryDefinition = styled.dd`
  text-align: right;
  margin: 0;
`

export const PurchaseSummaryDiscountList = styled(PurchaseSummaryList)`
  color: #ff7800;
`

export const PurchaseSummaryTotalList = styled(PurchaseSummaryList)`
  margin-top: 7px;
  font-weight: 700;
`
