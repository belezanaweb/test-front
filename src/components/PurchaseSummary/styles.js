import styled from 'styled-components'
import theme from '../../styles/theme'

export const Container = styled.div`
  border: 1px solid ${theme.colors.gray};
  border-radius: 3px;
  padding: ${theme.size[3]};
`

export const Item = styled.div`
  display: flex;
  justify-content: space-between;
  text-transform: uppercase;
  font-size: ${theme.size[3]};
  color: ${({ discount }) => (discount ? theme.colors.primaryLight : 'inhreit')};
  :not(:last-of-type) {
    margin-bottom: ${theme.spacing.xs};
  }
  :last-of-type {
    margin-top: ${theme.size[4]};
  }
  ${theme.media.desktop} {
    font-size: ${theme.size[4]};
  }
`
