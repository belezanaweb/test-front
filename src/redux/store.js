import { configureStore } from '@reduxjs/toolkit'
import productsReducer from './slices/productsSlice'
import summaryReducer from './slices/summarySlice'

export default configureStore({
  reducer: {
    products: productsReducer,
    summary: summaryReducer
  }
})
