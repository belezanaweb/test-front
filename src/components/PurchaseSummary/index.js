import React from 'react'
import { useSelector } from 'react-redux'
import * as S from './styles'
import format from '../../helpers/utility'

const PurchaseSummary = () => {
  const { cartData } = useSelector((state) => state.cart)

  return (
    <S.Container>
      <S.Item>
        <span>Produtos</span>
        <span>{format.price(cartData?.subTotal)}</span>
      </S.Item>

      <S.Item>
        <span>Frete</span>
        <span>{format.price(cartData?.shippingTotal)}</span>
      </S.Item>

      <S.Item discount>
        <span>Desconto</span>
        <span>{`- ${format.price(cartData?.discount)}`}</span>
      </S.Item>

      <S.Item>
        <strong>Total</strong>
        <strong>{format.price(cartData?.total)}</strong>
      </S.Item>
    </S.Container>
  )
}

export default PurchaseSummary
