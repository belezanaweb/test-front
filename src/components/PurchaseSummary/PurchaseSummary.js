import React, { useState, useEffect, useContext } from 'react'
import { CartContext } from '../../store/CartStore'
import {
  PurchaseSummaryWrapper,
  PurchaseSummaryList,
  PurchaseSummaryTerm,
  PurchaseSummaryDefinition,
  PurchaseSummaryDiscountList,
  PurchaseSummaryTotalList
} from './PurchaseSummary.styles'
import { Value } from '../Value/Value'

export const PurchaseSummary = () => {
  const cartContext = useContext(CartContext)
  const [summary, setSummary] = useState({})

  useEffect(() => {
    /* istanbul ignore else */
    if (cartContext.state.items) {
      const { subTotal, shippingTotal, discount, total } = cartContext.state

      setSummary({
        subTotal,
        shippingTotal,
        discount,
        total
      })
    }
  }, [cartContext])

  return (
    <PurchaseSummaryWrapper>
      <PurchaseSummaryList>
        <PurchaseSummaryTerm>Produtos</PurchaseSummaryTerm>
        <PurchaseSummaryDefinition>
          <Value>{summary.subTotal}</Value>
        </PurchaseSummaryDefinition>
      </PurchaseSummaryList>
      <PurchaseSummaryList>
        <PurchaseSummaryTerm>Frete</PurchaseSummaryTerm>
        <PurchaseSummaryDefinition>
          <Value>{summary.shippingTotal}</Value>
        </PurchaseSummaryDefinition>
      </PurchaseSummaryList>
      <PurchaseSummaryDiscountList>
        <PurchaseSummaryTerm>Desconto</PurchaseSummaryTerm>
        <PurchaseSummaryDefinition>
          &minus; <Value>{summary.discount}</Value>
        </PurchaseSummaryDefinition>
      </PurchaseSummaryDiscountList>
      <PurchaseSummaryTotalList>
        <PurchaseSummaryTerm>Total</PurchaseSummaryTerm>
        <PurchaseSummaryDefinition>
          <Value>{summary.total}</Value>
        </PurchaseSummaryDefinition>
      </PurchaseSummaryTotalList>
    </PurchaseSummaryWrapper>
  )
}
