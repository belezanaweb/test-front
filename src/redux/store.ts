import { createStore, applyMiddleware, combineReducers, compose } from 'redux'
import createSagaMiddleware from 'redux-saga'
import { composeWithDevTools } from 'redux-devtools-extension'

import AllReducers from './reducers'
import getSagas from './sagas'

const sagaMiddleware = createSagaMiddleware()
const composeEnhancers = composeWithDevTools({}) || compose

const store = createStore(
  combineReducers(AllReducers),
  composeEnhancers(applyMiddleware(sagaMiddleware))
)

sagaMiddleware.run(getSagas())

export type RootState = ReturnType<typeof store.getState>

export default store
