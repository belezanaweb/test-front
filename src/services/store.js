import { createStore } from 'redux';
import appReducer from './appReducer';

export default createStore(appReducer);