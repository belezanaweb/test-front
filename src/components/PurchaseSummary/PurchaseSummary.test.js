import React from 'react'
import { render } from '@testing-library/react'
import { CartStore } from '../../store/CartStore'
import { PurchaseSummary } from './PurchaseSummary'

const CartStoreValue = {
  state: {
    items: [],
    subTotal: 123.45,
    total: 432.1,
    shippingTotal: 345.67,
    discount: 37.02
  }
}

test('renders PurchaseSummary', () => {
  const { container } = render(
    <CartStore value={CartStoreValue}>
      <PurchaseSummary />
    </CartStore>
  )

  expect(container).toMatchSnapshot()
})
