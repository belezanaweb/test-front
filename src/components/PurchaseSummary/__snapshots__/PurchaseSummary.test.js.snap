// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PurchaseSummary 1`] = `
.c0 {
  display: grid;
  grid-gap: 9px;
  padding: 15px 13px 13px 14px;
  border: 1px solid #ccc;
  border-radius: 3px;
  color: #212122;
  font-size: 14px;
  line-height: 1.21428571428571;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin: 0;
}

.c2 {
  text-transform: uppercase;
}

.c3 {
  text-align: right;
  margin: 0;
}

.c4 {
  color: #ff7800;
}

.c5 {
  margin-top: 7px;
  font-weight: 700;
}

<div>
  <section
    class="c0"
  >
    <dl
      class="c1"
    >
      <dt
        class="c2"
      >
        Produtos
      </dt>
      <dd
        class="c3"
      >
        <span>
          R$ 123.45
        </span>
      </dd>
    </dl>
    <dl
      class="c1"
    >
      <dt
        class="c2"
      >
        Frete
      </dt>
      <dd
        class="c3"
      >
        <span>
          R$ 345.67
        </span>
      </dd>
    </dl>
    <dl
      class="c1 c4"
    >
      <dt
        class="c2"
      >
        Desconto
      </dt>
      <dd
        class="c3"
      >
        − 
        <span>
          R$ 37.02
        </span>
      </dd>
    </dl>
    <dl
      class="c1 c5"
    >
      <dt
        class="c2"
      >
        Total
      </dt>
      <dd
        class="c3"
      >
        <span>
          R$ 432.10
        </span>
      </dd>
    </dl>
  </section>
</div>
`;
